scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/SignatureAndHashAlgorithm.java

```diff
@@ -321,16 +322,18 @@ public static SignatureAndHashAlgorithm valueOf(String jcaName) {
 		if (index < 0) {
 			index = jcaName.indexOf("WITH");
 		}
+		HashAlgorithm hashAlgorithm;
+		SignatureAlgorithm signatureAlgorithm;
 		if (0 < index) {
 			String hash = jcaName.substring(0, index);
 			String signature = jcaName.substring(index + 4, jcaName.length());
-			HashAlgorithm hashAlgorithm = HashAlgorithm.valueOf(hash);
-			SignatureAlgorithm signatureAlgorithm = SignatureAlgorithm.valueOf(signature);
-			if (hashAlgorithm != null && signatureAlgorithm != null) {
-				return new SignatureAndHashAlgorithm(hashAlgorithm, signatureAlgorithm);
-			}
+			hashAlgorithm = HashAlgorithm.valueOf(hash);
+			signatureAlgorithm = SignatureAlgorithm.valueOf(signature);
+		} else {
+			hashAlgorithm = HashAlgorithm.INTRINSIC;
+			signatureAlgorithm = SignatureAlgorithm.valueOf(jcaName);
 		}
-		return null;
+		return new SignatureAndHashAlgorithm(hashAlgorithm, signatureAlgorithm);
```

Comment:
For me, the new code now assumes, that all signature- and hash-algorithms are at least known.
I don't think, that is a too good idea. The master branch already contains a similar update. Any reasons to work out your own?

Reply:
Actully I just copied them from master: https://github.com/eclipse/californium/blob/95f7197750c72d21df1f31b65194671058f1de53/scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/SignatureAndHashAlgorithm.java#L332
and my IDE remind me that the `if (hashAlgorithm != null && signatureAlgorithm != null) ` check is unnecessary so I removed them.

Reply:
Interesting ... 
I also read the javadoc for enum `valueOf` this morning and found, that it return an `IllegalArgumentException`. I guess, that requires to be addressed and fixed also in the master ;-).

Reply:
Even value of won't throw, `org.eclipse.californium.scandium.config.DtlsConnectorConfig.Builder#setSupportedSignatureAlgorithms(java.lang.String...)` will still throw IllegalArgumentException if that `valueOf` return null, so I think it probably okay?

Reply:
The idea of returning `null` instead of throwing an exception was to support other peer's, which sends certificates with "unsupported signatures and algorithms", if the trusted CA is before that. It requires more time to check, what is really needed to achieve that.

Reply:
But algorithms are decoded in here (org.eclipse.californium.scandium.dtls.SignatureAlgorithmsExtension#fromExtensionDataReader), it's not using `valueOf` method.

Reply:
> It requires more time to check, what is really needed to achieve that.

But not now. Now I have to focus on the little things :-).

Reply:
I checked it, it's only used for configuration. 
I adapted the result according the standard `valueOf` functionallity.
See PR #1752

Reply:
Should I revert the change of `SignatureAndHashAlgorithm.valueOf` so there won't be conflict? 
And should I amend this commit and perform a force push or just add a normal commit?

Reply:
And the small problem. Where should I put the "RSA" constant, it's a small thing but get my itchy.

Reply:
> Where should I put the "RSA" constant

What's about [Asn1DerDecoder.RSA](https://github.com/eclipse/californium/blob/master/element-connector/src/main/java/org/eclipse/californium/elements/util/Asn1DerDecoder.java#L79)?

Reply:
> Should I revert the change of SignatureAndHashAlgorithm.valueOf so there won't be conflict?
> And should I amend this commit and perform a force push or just add a normal commit?

The number of changes rather small. Chose what ever you want to apply.